### DISCUSSION TITLE
faild to find uinque target for patch

### DISCUSSION THREAD

**Discussion by geoHeil** [fluxcd/flux2#3635] [Q&A]
Hi, new to flux I struggle getting started.

I have followed the example of:

https://github.com/jakubhajek/traefik-flux-gitops

and want to add my own static HTTPS certs (i.e. as a minimal change to see if I can get started with flux)

To accomplish this I have run:

```
brew install mkcert nss
mkcert -install

mkcert myproject.com "*.myproject.com" myproject.local localhost 127.0.0.1 ::1
kubectl create namespace devlocal
kubectl create secret generic myproject-https-secret --from-file=tls.crt=./myproject.com+5.pem --from-file=tls.key=./myproject.com+5-key.pem --namespace devlocal
```

to generate the certificates and install a locally trusted CA.

Then I have added my customization for traefik into the staging namespace https://github.com/jakubhajek/traefik-flux-gitops/tree/main/apps/staging/traefik such that:

`kustomization.yaml` looks like:

```
---
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
namespace: traefik-devlocal
resources:
  - namespace.yaml
  - ../../base/traefik

patchesStrategicMerge:
  - traefik-patch.yaml
  - default-tls-store.yaml

```

As you can see, only the last line has been appended/added

`default-tls-storage.yaml` looks like:

```
---
apiVersion: traefik.containo.us/v1alpha1
kind: TLSStore
metadata:
  name: default

spec:
  defaultCertificate:
    secretName: myproject-https-secret
```

When applying the customization I am stuck with:

```
Kustomization reconciliation failed: kustomize build failed: accumulating resources: accumulation err='accumulating resources from './traefik': read /tmp/kustomization-2612227192/apps/devlocal/traefik: is a directory': recursed accumulation of path '/tmp/kustomization-2612227192/apps/devlocal/traefik': no matches for Id TLSStore.v1alpha1.traefik.containo.us/default.[noNs]; failed to find unique target for patch TLSStore.v1alpha1.traefik.containo.us/default.[noNs]
```

In short:

```
no matches for Id TLSStore.v1alpha1.traefik.containo.us/default.[noNs];
failed to find unique target for patch TLSStore.v1alpha1.traefik.containo.us/default.[noNs]
```

Even when adding an explicit namespace argument - the uniqueness constraint is still invalidated and the deployment fails.
